sanskrit_parser/base/sanskrit_base.py

```diff
@@ -84,15 +97,21 @@ def transcoded(self, encoding=None):
         """
         return sanscript.transliterate(self.thing, self.encoding, encoding)
 
-    def canonical(self):
+    def canonical(self, strict_io=True):
```

Comment:
Since these are instance methods, they could use the `strict_io` value provided during class __init__.

Reply:
@kmadathil 's comment applies here too. We create objects as and when needed in the graphs, sandhi tools, etc. We would need to propagate strict_io from the very first `__init__` to everywhere we create SanskritObject. It would be simpler to let the caller determine how they want the output to be formatted.

Reply:
Got it. Thanks!